common/functions/src/scalars/maths/round.rs

```diff
@@ -0,0 +1,189 @@
+// limitations under the License.
+
+use std::cmp::Ordering;
+use std::fmt;
+
+use common_datavalues::columns::DataColumn;
+use common_datavalues::prelude::*;
+use common_datavalues::DataSchema;
+use common_datavalues::DataType;
+use common_exception::Result;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+struct RoundingFunction {
+    display_name: String,
+    rounding_func: fn(f64) -> f64,
+}
+
+impl RoundingFunction {
+    pub fn try_create(display_name: &str, f: fn(f64) -> f64) -> Result<Box<dyn Function>> {
+        Ok(Box::new(Self {
+            display_name: display_name.to_string(),
+            rounding_func: f,
+        }))
+    }
+}
+
+impl Function for RoundingFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        0
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        Some((1, 2))
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(true)
+    }
+
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::Float64)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let x_series = columns[0]
+            .column()
+            .to_minimal_array()?
+            .cast_with_type(&DataType::Float64)?;
+
+        let column: DataColumn = if columns.len() == 1 {
+            DFFloat64Array::new_from_opt_iter(
+                x_series
+                    .f64()?
+                    .iter()
+                    .map(|x| x.map(|x| (self.rounding_func)(*x))),
+            )
+        } else {
+            let d_column: &DataColumn = &columns[1].column().cast_with_type(&DataType::Int64)?;
+            match d_column {
+                DataColumn::Constant(v, _) => {
+                    if v.is_null() {
+                        // mysql> SELECT ROUND(1.015, NULL); => NULL
+                        DFFloat64Array::full_null(input_rows)
+                    } else {
+                        match v.as_i64() {
+                            Err(_) => {
+                                // mysql> SELECT ROUND(1.015, 'aaa'); => 1 => d == 0
+                                DFFloat64Array::new_from_opt_iter(
+                                    x_series
+                                        .f64()?
+                                        .iter()
+                                        .map(|x| x.map(|x| (self.rounding_func)(*x))),
+                                )
+                            }
+                            Ok(v) => match v.cmp(&0) {
+                                Ordering::Greater => {
+                                    let z = 10_f64.powi(if v > 30 { 30 } else { v as i32 });
+                                    DFFloat64Array::new_from_opt_iter(x_series.f64()?.iter().map(
```

Comment:
ArrayApply is better than `new_from_opt_iter`.

Reply:
Thanks, change it.